src/parameterizations/lateral/MOM_MEKE.F90

```diff
@@ -412,22 +415,64 @@ subroutine step_forward_MEKE(MEKE, h, SN_u, SN_v, visc, dt, G, GV, US, CS, hu, h
     !$OMP parallel do default(shared)
     do j=js,je ; do i=is,ie
       src(i,j) = CS%MEKE_BGsrc
-      src_adv(i,j) = 0.
-      src_mom_K4(i,j) = 0.
-      src_btm_drag(i,j) = 0.
-      src_GM(i,j) = 0.
-      src_mom_lp(i,j) = 0.
-      src_mom_bh(i,j) = 0.
     enddo ; enddo
 
+    ! Initialize diagnostics
+    if (CS%id_src_adv > 0) src_adv(is:ie, js:je) = 0.
+    if (CS%id_src_GM > 0) src_GM(is:ie, js:je) = 0.
+    if (CS%id_src_mom_lp > 0) src_mom_lp(is:ie, js:je) = 0.
+    if (CS%id_src_mom_bh > 0) src_mom_bh(is:ie, js:je) = 0.
+    if (CS%id_src_mom_K4 > 0) src_mom_K4(is:ie, js:je) = 0.
+    if (CS%id_src_btm_drag > 0) src_btm_drag(is:ie, js:je) = 0.
+
+    ! Identify any damped diagnostics in first stage of Strang splitting
+    any_damping_diags_s1 = any([ &
+        CS%id_src_GM > 0, &
+        CS%id_src_mom_lp > 0, &
+        CS%id_src_mom_bh > 0, &
+        CS%id_src_btm_drag > 0 &
+    ])
+
+    ! Identify any damped diagnostics
+    any_damping_diags = any([ &
+        any_damping_diags_s1, &
+        CS%id_src_adv > 0, &
+        CS%id_src_mom_K4 > 0 &
+    ])
+
     if (allocated(MEKE%mom_src)) then
       !$OMP parallel do default(shared)
       do j=js,je ; do i=is,ie
-        src(i,j) = src(i,j) - CS%MEKE_FrCoeff*I_mass(i,j)*MEKE%mom_src(i,j) &
-                   - (CS%MEKE_bhFrCoeff-CS%MEKE_FrCoeff)*I_mass(i,j)*MEKE%mom_src_bh(i,j)
-        src_mom_lp(i,j) = - CS%MEKE_FrCoeff*I_mass(i,j)*(MEKE%mom_src(i,j)-MEKE%mom_src_bh(i,j))
-        src_mom_bh(i,j) = - CS%MEKE_bhFrCoeff*I_mass(i,j)*MEKE%mom_src_bh(i,j)
+        src(i,j) = src(i,j) - CS%MEKE_FrCoeff * I_mass(i,j) * MEKE%mom_src(i,j)
+      enddo ; enddo
+    endif
+
+    if (allocated(MEKE%mom_src_bh)) then
+      if (CS%MEKE_bhFrCoeff > 0. .and. CS%MEKE_FrCoeff > 0.) then
+        bh_coeff = CS%MEKE_bhFrCoeff - CS%MEKE_FrCoeff
+      else
+        bh_coeff = CS%MEKE_bhFrCoeff
+      endif
+
+      !$OMP parallel do default(shared)
+      do j=js,je ; do i=is,ie
+        src(i,j) = src(i,j) - bh_coeff * I_mass(i,j) * MEKE%mom_src_bh(i,j)
       enddo ; enddo
+
+      if (CS%id_src_mom_lp > 0) then
```

Comment:
Would one ever register `src_mom_lp` if `MEKE_FrCoeff` were unset?  Maybe it's better to address this possibility at initialization, rather than here.

Reply:
Agree. We should only register src_mom_lp when MEKE_FrCoeff is set.

Reply:
I've moved the following diagnostics into conditional blocks:

* `id_src_mom_lp`
* `id_src_mom_bh`
* `id_src_GM`
* `id_src_mom_K4`